test/simulator/test/org/apache/cassandra/simulator/test/AccordJournalSimulationTest.java

```diff
@@ -18,250 +18,359 @@
+        public int serializedSize(String key, String value, int userVersion)
         {
-            for (int i = 0; i < State.events; i++)
-            {
-                TxnRequest<?> event = State.journal.readMessage(State.toTxnId(i), MessageType.PRE_ACCEPT_REQ, PreAccept.class);
-                State.logger.info("Event {} -> {}", i, event);
-                if (event == null)
-                    throw new AssertionError(String.format("Unable to read event %d", i));
-            }
-            State.logger.info("Check complete");
+            return TypeSizes.INT_SIZE + key.length();
         }
-        finally
+
+        @Override
+        public void serialize(String key, String value, DataOutputPlus out, int userVersion) throws IOException
         {
-            State.journal.shutdown();
+            out.writeInt(key.length());
+            out.writeBytes(key);
+        }
+
+        @Override
+        public String deserialize(String key, DataInputPlus in, int userVersion) throws IOException
+        {
+            int size = in.readInt();
+            byte[] value = new byte[size];
+            for (int i = 0; i < size; i++)
+                value[i] = in.readByte();
+
+            return new String(value);
+        }
+    }
+
+    @Isolated
+    public static class IdentityKeySerializer implements KeySupport<String>
+    {
+
+        @Override
+        public int serializedSize(int userVersion)
+        {
+            return 16;
+        }
+
+        @Override
+        public void serialize(String key, DataOutputPlus out, int userVersion) throws IOException
+        {
+            int maxSize = 16 - TypeSizes.INT_SIZE;
+            if (key.length() > maxSize)
+                throw new IllegalStateException();
+
+            out.writeInt(key.length());
+            out.writeBytes(key);
+            int remaining = maxSize - key.length();
+            for (int i = 0; i < remaining; i++)
+                out.writeByte(0);
+        }
+
+        @Override
+        public String deserialize(DataInputPlus in, int userVersion) throws IOException
+        {
+            int size = in.readInt();
+            byte[] key = new byte[size];
+            for (int i = 0; i < size; i++)
+                key[i] = in.readByte();
+
+            int maxSize = 16 - TypeSizes.INT_SIZE;
+            int remaining = maxSize - size;
+            for (int i = 0; i < remaining; i++)
+                in.readByte(); // todo assert 0
+
+            return new String(key);
+        }
+
+        @Override
+        public String deserialize(ByteBuffer buffer, int position, int userVersion)
+        {
+            int size = buffer.getInt();
+            byte[] key = new byte[size];
+            for (int i = 0; i < size; i++)
+                key[i] = buffer.get();
+
+            int maxSize = 16 - TypeSizes.INT_SIZE;
+            int remaining = maxSize - size;
+            for (int i = 0; i < remaining; i++)
+                buffer.get(); // todo assert 0
```

Comment:
want to fix the todo?  Also `0` is common so could be good to use another value like `4` (randomly selected from XKCD)